plugins/flutter_plugins/url_launcher_fde/linux/url_launcher_plugin.cc

```diff
@@ -44,6 +44,11 @@ class UrlLauncherPlugin : public flutter::Plugin {
       std::unique_ptr<flutter::MethodResult<EncodableValue>> result);
 };
 
+// utility function to compare url schemes
+static bool startsWith(const std::string& s, const std::string& prefix) {
+    return s.size() >= prefix.size() && s.compare(0, prefix.size(), prefix) == 0;
```

Comment:
You don't need the first half of this check; compare will return non-zero if `s` is shorter than `prefix`.